tools/flow_grid.py

```diff
@@ -0,0 +1,689 @@
+import numpy as np
+import pandas as pd
+import sys
+import ast
+
+class flow_grid(object):        
+    """
+    Container class for holding and manipulating gridded VIC routing data.
+    Can be instantiated with optional keyword arguments. These keyword
+    arguments will be used to add a dataset (dem, flowdir, accumulation)
+    to the flow_grid instance.
+
+    Parameters
+    ----------
+    data : string or numpy ndarray (optional)
+           Data to be read. Can either be a file name or an array.
+           If data is from a file, 'input_type' should be set to the
+           appropriate value ('ascii' or 'raster').
+    data_type : 'dem', 'dir', 'acc' (optional)
+                 How to interpret the input data:
+                     'dem' : digital elevation data
+                     'dir' : flow direction data
+                     'acc' : flow accumulation (upstream area) data
+                
+    input_type : 'raster', 'ascii' or 'array' (optional)
+                 Type of input data.
```

Comment:
If an unknown string is passed to datatype, then the data will be added under an attribute name specified by that string. For instance, if you passed 'alpha', it would be added as `self.alpha`. This is useful if you need extra parameter files aside from the defaults (such as Mohseni parameter files for the VIC-RBM model). The defaults ('dem' and 'dir') are used internally by class methods, which is why I included them in the docstring.

Input type only takes those three named values though.

I'll update the docstring to make this more clear.